torchaudio/prototype/tacotron2.py

```diff
@@ -0,0 +1,844 @@
+        attention_context = torch.bmm(attention_weights.unsqueeze(1), memory)
+        attention_context = attention_context.squeeze(1)
+
+        return attention_context, attention_weights
+
+
+class _Prenet(nn.Module):
+    r"""Prenet Module. It is consists of ``len(output_size)`` linear layers.
+
+    Args:
+        in_dim (int): The size of each input sample.
+        output_sizes (list): The output dimension of each linear layers.
+    """
+
+    def __init__(self, in_dim: int, out_sizes: List[int]) -> None:
+        super().__init__()
+        in_sizes = [in_dim] + out_sizes[:-1]
+        self.layers = nn.ModuleList(
+            [_get_linear_layer(in_size, out_size, bias=False)
+             for (in_size, out_size) in zip(in_sizes, out_sizes)])
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""Pass the input through Prenet.
+
+        Args:
+            x (Tensor): The input sequence to Prenet with shape (n_batch, in_dim).
+
+        Return:
+            x (Tensor): Tensor with shape (n_batch, sizes[-1])
+        """
+
+        for linear in self.layers:
+            x = F.dropout(F.relu(linear(x)), p=0.5, training=True)
+        return x
+
+
+class _Postnet(nn.Module):
+    r"""Postnet Module.
+
+    Args:
+        n_mel (int): Number of mel bins.
+        postnet_embedding_dim (int): Postnet embedding dimension.
+        postnet_kernel_size (int): Postnet kernel size.
+        postnet_n_convolution (int): Number of postnet convolutions.
+    """
+
+    def __init__(self, n_mel: int, postnet_embedding_dim: int,
+                 postnet_kernel_size: int, postnet_n_convolution: int):
+        super().__init__()
+        self.convolutions = nn.ModuleList()
+
+        self.convolutions.append(
+            nn.Sequential(
+                _ConvNorm(n_mel, postnet_embedding_dim,
+                          kernel_size=postnet_kernel_size, stride=1,
+                          padding=int((postnet_kernel_size - 1) / 2),
+                          dilation=1, w_init_gain='tanh'),
+                nn.BatchNorm1d(postnet_embedding_dim))
+        )
+
+        for _ in range(1, postnet_n_convolution - 1):
+            self.convolutions.append(
+                nn.Sequential(
+                    _ConvNorm(postnet_embedding_dim,
+                              postnet_embedding_dim,
+                              kernel_size=postnet_kernel_size, stride=1,
+                              padding=int((postnet_kernel_size - 1) / 2),
+                              dilation=1, w_init_gain='tanh'),
+                    nn.BatchNorm1d(postnet_embedding_dim))
+            )
+
+        self.convolutions.append(
+            nn.Sequential(
+                _ConvNorm(postnet_embedding_dim, n_mel,
+                          kernel_size=postnet_kernel_size, stride=1,
+                          padding=int((postnet_kernel_size - 1) / 2),
+                          dilation=1, w_init_gain='linear'),
+                nn.BatchNorm1d(n_mel))
+        )
```

Comment:
Combining these blocks is more readable. 

```python
for i in range(postnet_n_convolution):
  in_channels = n_mel if i == 0 else postnet_embedding_dim
  out_channels = n_mel if i == postnet_n_convolution - 1 else postnet_embedding_dim
  init_gain = 'linear' if i == postnet_n_convolution - 1 else 'tanh'
  self.convolutions.append(
            nn.Sequential(
                _ConvNorm(in_channels, out_channels,
                          kernel_size=postnet_kernel_size, stride=1,
                          padding=int((postnet_kernel_size - 1) / 2),
                          dilation=1, w_init_gain='linear'),
                nn.BatchNorm1d(n_mel)
             )
        )
```

I'd also prefer a way of removing the sequential, but giving how it is used below, I'd be ok keeping it as is.

Reply:
Thanks. Addressed [here](https://github.com/pytorch/audio/pull/1621/files#diff-8b8ec57192b9aaeb99a9ec5e709c977771b31b49fe7ddc6704c2128cf161a00eR392)